### ISSUE TITLE
loader.loaderId is not a function

### ISSUE THREAD

**Issue by baumandm** [dotansimha/graphql-code-generator#6374]
**Describe the bug**
 

After upgrading from Node 14.17.3 to Node 16.5.0, the following command started failing for me:

```
❯ npx graphql-codegen --config codegen.yml
loader.loaderId is not a function
```

Although if I remove the explicit config location, it auto-detects my config and works:

```
❯ npx graphql-codegen
  ✔ Parse configuration
  ✔ Generate outputs
```

**To Reproduce**

I reproduced it with the default codesandbox template:

https://codesandbox.io/s/priceless-roentgen-rc0kf?file=/package.json

`yarn codegen` - Fails with reported issue :x:
`yarn codegen-alt` - Works as expected ✔️ 

**Environment:**

- OS: MacOS
- "@graphql-codegen/cli": "1.21.5",
- "@graphql-codegen/introspection": "^1.18.2",
- "@graphql-codegen/typescript": "1.22.3",
- "@graphql-codegen/typescript-operations": "1.18.2",
- NodeJS: 16.x

**Additional context**
Reproduces with both yarn and npm.

**Comment by dotansimha** [comment]
I think some dependencies resolves to v8 of graphql-tools, which has some breaking changes. I'm trying to check why. 

fyi @ardatan

**Comment by florentin-soubrand** [comment]
I noticed personally if you remove '--config' it takes autmaticaly the codegen.yml file and no more issue appear

**Comment by MidnightDesign** [comment]
> I noticed personally if you remove '--config' it takes autmaticaly the codegen.yml file and no more issue appear

Doesn't change anything for me. Still getting the same error.

**Comment by dotansimha** [comment]
The issue seems to be related to integration with `graphql-config` and version resolution. 
We are working on a fix and we hope to release a hotfix soon :)

**Comment by riccardocaranfil** [comment]
+1 here but got this error also running the command without passing the config

**Comment by dotansimha** [comment]
### Background

GraphQL-Tools v8 was released a few hours ago, with many fixes, and some breaking changes. Codegen heavily depends on graphql-tools loaders to load the schema and documents. 

One of the packages (`@graphql-tools/prisma-loader`) missed a version bump and got a release for `6.3.1`. Codegen depends at the moment on `^6.3.0` so new projects that just installed codegen got `6.3.1` with the breaking change, leading to a conflict while `graphql-config` was running. 

#### Solution

A fix was issued in this PR: https://github.com/dotansimha/graphql-code-generator/pull/6375

UPDATE: Releasing in a few minutes :) 

### Workaround:

If you are not able to upgrade to latest version of codegen, from any reason, you can still solve this issue locally.

#### With `yarn`

If you are using Yarn, you can pin the version using resolutions, add this to your `package.json` and reinstall the dependencies:

```json
  "resolutions": {
    "@graphql-tools/prisma-loader": "6.3.0"
  }
```

If you are still having this issue, try to run `yarn why @graphql-tools/prisma-loader` and make sure it's pinned to `6.3.0`. If you got `6.3.1` - try to clear dependencies and try again. 

#### With `npm` 

The latest version of NPM does hoisting, you can install `@graphql-tools/prisma-loader@6.3.0` in your project this should avoid the hoisting:

```
npm install --save-dev @graphql-tools/prisma-loader@6.3.0
```

If you are still having this issue, try to run `npm ls @graphql-tools/prisma-loader` and make sure it's pinned to `6.3.0`. If you got `6.3.1` - try to clear dependencies and try again. 

#### Next Steps

We are really excited with the new v8 release of `graphql-tools`, and we have an ongoing PR to upgrade it: https://github.com/dotansimha/graphql-code-generator/pull/6373 . For most codegen users it should change anything, maybe just some performance improvements :)

**Comment by calirvine** [comment]
Thanks for the quick fix, just  installed the alpha branch for `1.21.8` for now and it works as expected .

**Comment by dotansimha** [comment]
Fixed in `@graphql-codegen/cli@1.21.8` :)

**Comment by mheob** [comment]
Works well again now. Thank you for the very quick fix.
